Add initial challenge template (#10)
model-checking/verify-rust-std: doc/src/template.md

# Challenge XXXX[^challenge_id]: Challenge Title

- **Status:** *One of the following: [Open | Resolved | Expired]*
- **Solution:** *Option field to point to the PR that solved this challenge.*
- **Tracking Issue:** *Link to issue*
- **Start date:** *YY/MM/DD*
- **End date:** *YY/MM/DD*

-------------------

## Goal

*Describe the goal of this challenge with 1-2 sentences.*

## Motivation

*Explain why this is a challenge that should be prioritized. Consider using a motivating example.*

## Description

*Describe the challenge in more details.*

### Assumptions

*Mention any assumption that users may make. Example, "assuming the usage of Stacked Borrows".*

### Success Criteria*

> *Here are few possible criteria:*

Reviewer: How about "Here are some examples of possible criteria:" (else at least insert "a" before "few possible")

> *Here are few possible criteria:*

Author: ```suggestion
*Here are some examples of possible criteria:*
```

All the following unsafe functions must be annotated with safety contracts and the contracts have been verified:

|Function	|Location	|
|---	|---	|
|	|	|
|	|	|
|	|	|
|	|	|
|	|	|

At least N of the following usages were proven safe:

|Function	|Location	|
|---	|---	|
|	|	|
|	|	|
|	|	|
|	|	|
|	|	|

All proofs must automatically ensure the absence of the following undefined behaviors [[ref]](https://github.com/rust-lang/reference/blob/142b2ed77d33f37a9973772bd95e6144ed9dce43/src/behavior-considered-undefined.md):

*List of UBs*

Note: All solutions to verification challenges need to satisfy the criteria established in the [challenge book](general-rules.md)

> in addition to the ones listed above

Reviewer: Add a full stop at the end of the sentence.

Author: ```suggestion
in addition to the ones listed above.
```

> # Challenge XXXX[^challenge_id]: Challenge Title

Reviewer: Is "Challenge" the right wording?

Author: Any suggestions? We could use `Verification Task` or something like that.

- **Status:** *One of the following: [Open | Resolved | Expired]*

>   - **Solution:** *Option field to point to the PR that solved this challenge.*

Reviewer: Is this indentation intentional?

Author: ```suggestion
- **Solution:** *Option field to point to the PR that solved this challenge.*
```

- **Tracking Issue:** *Link to issue*
- **Start date:** *YY/MM/DD*
- **End date:** *YY/MM/DD*

-------------------

Status: [Open | Resolved | Expired]:
Tracking issue:
Start date:
Due date:

> Author(s):

Reviewer: This somehow overlaps with the lines above?!

Author: My bad. :)

> 
> 
> Status: [Open | Resolved | Expired]:
> Tracking issue:
> Start date:
> Due date:
> Author(s):

Author: ```suggestion
```

## Goal

*Describe the goal of this challenge with 1-2 sentences.*

## Motivation

*Explain why this is a challenge that should be prioritized. Consider using a motivating example.*

## Description

*Describe the challenge in more details.*

### Assumptions

> *Mention any assumption that users may make. Example, "assuming the usage of Stack Borrow".*

Reviewer: s/Stack Borrow/Stacked Borrows/

Author: Why do I keep making the same mistake? :disappointed:

> *Mention any assumption that users may make. Example, "assuming the usage of Stack Borrow".*

Author: ```suggestion
*Mention any assumption that users may make. Example, "assuming the usage of Stacked Borrows".*
```

### Success Criteria*

*Here are few possible criteria:*

All the following unsafe functions must be annotated with safety contracts and the contracts have been verified:

|Function	|Location	|
|---	|---	|
|	|	|
|	|	|
|	|	|
|	|	|
|	|	|

At least N of the following usages were proven safe:

|Function	|Location	|
|---	|---	|
|	|	|
|	|	|
|	|	|
|	|	|
|	|	|

All proofs must automatically ensure the absence of the following undefined behaviors [[ref]](https://github.com/rust-lang/reference/blob/142b2ed77d33f37a9973772bd95e6144ed9dce43/src/behavior-considered-undefined.md):

*List of UBs*

Note: All solutions to verification challenges need to satisfy the criteria established in the [challenge book](general-rules.md)

> in addition to the ones listed below

Reviewer: What is listed below?!

Author: ```suggestion
in addition to the ones listed above
```
:smile:
